packages/pages/src/dev/server/ssr/getLocalData.ts

```diff
@@ -74,21 +74,23 @@ export const getLocalDataManifest = async (
       continue;
     }
 
+    const templateModuleInternal = await findTemplateModuleInternal(
+      vite,
+      async (t) => featureName === t.config.name,
+      templateFilepaths
+    );
+
     const uid = data.uid?.toString();
     const entityId = data.id?.toString();
-    const slug = data.slug?.toString();
+    const slugField = templateModuleInternal?.config?.slugField;
+    const slug = slugField ? data[slugField] : data.slug?.toString();
```

Comment:
Do you need to .toString() data[slugField]? Or is toString not necessary on data.slug?

Reply:
We shouldn't to toString on `data[slugField]` so that it is properly passed as undefined in the case that it isn't there.